## Project - France

In [140]:
import pandas as pd
import numpy as np
import plotly.express as px
regions = pd.read_csv("./Data/noc_regions.csv")
athlete_events = pd.read_csv("./Data/athlete_events.csv")

In [3]:
regions.head(3)

NOC       region                 notes
0  AFG  Afghanistan                   NaN
1  AHO      Curacao  Netherlands Antilles
2  ALB      Albania                   NaN

- a. hur många länder som är med i datan?

In [4]:
print(f"{regions['NOC'].nunique() = }")
print(f"{regions['region'].nunique() = }")
print(f"{athlete_events['NOC'].nunique() = }")
print(f"{athlete_events['Team'].nunique() = }")

regions['NOC'].nunique() = 230
regions['region'].nunique() = 206
athlete_events['NOC'].nunique() = 230
athlete_events['Team'].nunique() = 1184


Svar: Det finns 206 'region' representerade i datasetet och det finns 230 "NOC"'s (National Olympic Committée's) representerade. Att det är olika antal beror på att samma 'region' historiskt har haft flera olika nationella olympiska komittéer.
Några exepel:
- 'region' som heter 'Germany' har historiskt varit representerade av flera NOC's: Väst Tyskland (FRG), Östtyskland (GDR) och Saar Protectorate (SAA).
- 'region' som heter 'Russia' har historiskt varit representerade av flera NOC's: Soviet Union (URS), Unified Team (EUN).
- Det finns 1184 'Team' representerade i datasetet. 
  
Slutsats: Man kan säga att det finns 230 länder representerade, eftersom det finns 230 NOC's representerade. Men det är en definitionsfråga.
  
[Link to Wikipedia](https://en.wikipedia.org/wiki/All-time_Olympic_Games_medal_table)

In [5]:
display(regions[regions['region'] == 'Germany'])
display(regions[regions['region'] == 'Russia'])

NOC   region notes
70   FRG  Germany   NaN
76   GDR  Germany   NaN
79   GER  Germany   NaN
173  SAA  Germany   NaN

NOC  region notes
66   EUN  Russia   NaN
171  RUS  Russia   NaN
214  URS  Russia   NaN

In [6]:
athlete_events.head(3)


ID                 Name Sex   Age  Height  Weight     Team  NOC  \
0   1            A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   2             A Lamusi   M  23.0   170.0    60.0    China  CHN   
2   3  Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   

         Games  Year  Season       City       Sport  \
0  1992 Summer  1992  Summer  Barcelona  Basketball   
1  2012 Summer  2012  Summer     London        Judo   
2  1920 Summer  1920  Summer  Antwerpen    Football   

                          Event Medal  
0   Basketball Men's Basketball   NaN  
1  Judo Men's Extra-Lightweight   NaN  
2       Football Men's Football   NaN

- b. vilka länder är med? (förkortningarna räcker)

In [7]:
unique_regions = regions['region'].unique() # skapar en variabel ifall vi behöver den senare.
NOCs = pd.Index(regions['NOC'].unique())
print(f"Dessa National Olympic Commités är med i datasetet: {NOCs}.")

Dessa National Olympic Commités är med i datasetet: Index(['AFG', 'AHO', 'ALB', 'ALG', 'AND', 'ANG', 'ANT', 'ANZ', 'ARG', 'ARM',
       ...
       'VIE', 'VIN', 'VNM', 'WIF', 'YAR', 'YEM', 'YMD', 'YUG', 'ZAM', 'ZIM'],
      dtype='object', length=230).


-  c. vilka sporter är med?

In [8]:
#sports = pd.Index(athlete_events['Sport'].sort_values().unique())
sports = athlete_events['Sport'].sort_values().unique()
#type(sports)
"Aeronautics" in sports
print(f"Dessa sporter är med: {sports}.")

Dessa sporter är med: ['Aeronautics' 'Alpine Skiing' 'Alpinism' 'Archery' 'Art Competitions'
 'Athletics' 'Badminton' 'Baseball' 'Basketball' 'Basque Pelota'
 'Beach Volleyball' 'Biathlon' 'Bobsleigh' 'Boxing' 'Canoeing' 'Cricket'
 'Croquet' 'Cross Country Skiing' 'Curling' 'Cycling' 'Diving'
 'Equestrianism' 'Fencing' 'Figure Skating' 'Football' 'Freestyle Skiing'
 'Golf' 'Gymnastics' 'Handball' 'Hockey' 'Ice Hockey' 'Jeu De Paume'
 'Judo' 'Lacrosse' 'Luge' 'Military Ski Patrol' 'Modern Pentathlon'
 'Motorboating' 'Nordic Combined' 'Polo' 'Racquets' 'Rhythmic Gymnastics'
 'Roque' 'Rowing' 'Rugby' 'Rugby Sevens' 'Sailing' 'Shooting'
 'Short Track Speed Skating' 'Skeleton' 'Ski Jumping' 'Snowboarding'
 'Softball' 'Speed Skating' 'Swimming' 'Synchronized Swimming'
 'Table Tennis' 'Taekwondo' 'Tennis' 'Trampolining' 'Triathlon'
 'Tug-Of-War' 'Volleyball' 'Water Polo' 'Weightlifting' 'Wrestling'].


-  d. vilka medaljtyper finns det?

In [9]:
medal_types = pd.Index(athlete_events['Medal'].unique())
medal_types = medal_types.dropna()
print(f'The medal types are {medal_types[0]}, {medal_types[1]} and {medal_types[2]}.')

The medal types are Gold, Bronze and Silver.


-  e. ta reda på statistik för åldern: medelvärde, median, min, max, standardavvikelse,

In [10]:
# age_statistics = athlete_events["Age"].agg(["mean","median","min","max","std"])
# age_statistics
athlete_events['Age'].describe()

count    261642.000000
mean         25.556898
std           6.393561
min          10.000000
25%          21.000000
50%          24.000000
75%          28.000000
max          97.000000
Name: Age, dtype: float64

### f. utforska datan vidare med egna frågor.
- Vilket land har fått flest guldmedaljer (Arsen)
- 10 bästa länderna (om hinner: uppdelat på sommar och vinter OS) (Arsen)
- Medelålder, högsta och lägsta ålder för medaljörer. (Asajad)
- Atleter med flest medaljer
- Mest populära sporter
- Atleter som vunnit i fler än en sport

In [34]:
# 0f, kod här:

In [91]:

# get all gold medalists
gold_medalists = athlete_events.groupby('Medal').get_group('Gold')
# get all silver medalists
silver_medalists = athlete_events.groupby('Medal').get_group('Silver')
# get all bronze medalists
bronze_medalists = athlete_events.groupby('Medal').get_group('Bronze')

# print the average age of gold medalists, the oldest gold medalist's age and the youngest gold medalist's age
print(f'The average age of Gold medalists is: {gold_medalists["Age"].mean():.1f}')
print(f'The oldest Gold medalist was {gold_medalists["Age"].max():.0f} years old.')
print(f'The youngest Gold medalist was {gold_medalists["Age"].min():.0f} years old.')
print()

# print the average age of silver medalists, the oldest silver medalist's age and the youngest silver medalist's age
print(f'The average age of Silver medalists is: {silver_medalists["Age"].mean():.1f}')
print(f'The oldest Silver medalist was {silver_medalists["Age"].max():.0f} years old.')
print(f'The youngest Silver medalist was {silver_medalists["Age"].min():.0f} years old.')
print()

# print the average age of bronze medalists, the oldest bronze medalist's age and the youngest bronze medalist's age
print(f'The average age of Bronze medalists is: {bronze_medalists["Age"].mean():.1f}')
print(f'The oldest Bronze medalist was {bronze_medalists["Age"].max():.0f} years old.')
print(f'The youngest Bronze medalist was {bronze_medalists["Age"].min():.0f} years old.')

The average age of Gold medalists is: 25.9
The oldest Gold medalist was 64 years old.
The youngest Gold medalist was 13 years old.

The average age of Silver medalists is: 26.0
The oldest Silver medalist was 73 years old.
The youngest Silver medalist was 11 years old.

The average age of Bronze medalists is: 25.9
The oldest Bronze medalist was 72 years old.
The youngest Bronze medalist was 10 years old.


In [83]:
# select only rows where 'Medal' column is not null
athlete_medalists = athlete_events.dropna(subset=['Medal'])

# get the name of the athlete with the most medals
athlete_medalists_name = athlete_medalists["Name"].value_counts().index[0]

# get the count of medals for each athlete
athlete_medalists_name_count = athlete_events["Name"].value_counts()

# get the count of medals for the athlete with the most medals
highest_medals_count = athlete_medalists_name_count.iloc[0]

# print the name of the athlete with the most medals and the count of their medals
print(f'{athlete_medalists_name} has the highest number of medals with {highest_medals_count} medals.')


Michael Fred Phelps, II has the highest number of medals with 28 medals.


In [52]:
most_common_sport = athlete_events['Sport'].value_counts().index[0]
print(f"The most common sport is {most_common_sport}.")


The most common sport is Athletics.


In [90]:
# create a new dataframe with only the relevant columns
athlete_medals = athlete_events[['Name', 'Sport', 'Medal']].dropna()

# group the dataframe by 'Name' and 'Sport', and count the number of unique 'Medal' values
athlete_medals_grouped = athlete_medals.groupby(['Name', 'Sport'])['Medal'].nunique()

# filter the dataframe to only include rows where the count of unique 'Medal' values is greater than 1
athlete_medals_filtered = athlete_medals_grouped[athlete_medals_grouped > 2]

# display the filtered dataframe
athlete_medals_filtered.info()

<class 'pandas.core.series.Series'>
MultiIndex: 689 entries, ('Abdul Rashid', 'Hockey') to ('va Grard-Novk', 'Swimming')
Series name: Medal
Non-Null Count  Dtype
--------------  -----
689 non-null    int64
dtypes: int64(1)
memory usage: 228.3+ KB


### 0f. Om vi har tid:
- Karta med alla länder som varit med i sommar-OS och vinter-OS, over time med animation frame.

In [13]:
# 0f, kartor, kod här:

### g: Diagram över könsfördelningen (Daniel)
- könsfördelning per OS över tid

In [175]:
# 0g kod här:

male_athletes = athlete_events.groupby("Sex").get_group("M")
female_athletes = athlete_events.groupby("Sex").get_group("F")

female_athletes_by_year_unique_count = female_athletes.groupby("Year")["Name"].nunique()
female_athletes_by_year_unique_count.loc[1896] = 0
female_athletes_by_year_unique_count = female_athletes_by_year_unique_count.sort_index()

male_athletes_by_year_unique_count = male_athletes.groupby("Year")["Name"].nunique()

# create a dataframe with the data
female_male_ratio_over_time_df = pd.DataFrame({'Year': male_athletes_by_year_unique_count.index, 'Male Athletes': male_athletes_by_year_unique_count.values, 'Female Athletes': female_athletes_by_year_unique_count.values})

# plot the data using Plotly Express
female_male_ratio_over_time_plotly = px.line(female_male_ratio_over_time_df, x='Year', y=['Male Athletes', 'Female Athletes'], title='Male and Female Athletes Count Over Time')
female_male_ratio_over_time_plotly.show()






###  h. diagram över topp 10 länder som tagit flest medaljer:
- Arsen

In [13]:
# 0h kod här:

### i. plotta gärna fler saker som är intressant:


In [14]:
# 0i kod här:
